### PR TITLE
Merging Tulane Fork into Main

### PR DESCRIPTION
**Pull request by outlawhayden**
Hey all -

I'm opening this I think to begin the process of folding our changes into your main system. A few days ago, I tried just pulling everything in from your end onto ours first before opening this and it went pretty poorly - you can see what happened [here](https://github.com/eye-on-surveillance/sawt/commit/b2acc6b2e5d43871b673e803ada7350e68288532). It says these branches can be automatically merged, but I think we shouldn't just do that, because I did a revert on that massive merge I tried, and I think that's messing with it and I don't code to just get obliterated for no reason.

I think most of the changes on your end can just pave over ours, especially with respect to the config files, AI stuff, etc., - the big concerns I think are the new web components we added, and a couple of functions we made in getanswer that allow us to iterate through the getanswer function and get responses for multiple k values and package them together.

### CODE REVIEW

File: packages/googlecloud/functions/getanswer/main.py
Comment (outlawhayden) [inline packages/googlecloud/functions/getanswer/main.py:1]: one of the main areas i was not sure how to integrate. the version we were working with did not have the supabase client at this stage, and we were just poking around with a local deploy

File: packages/web/components/BetaCard.tsx
Comment (outlawhayden) [inline packages/web/components/BetaCard.tsx:1]: where we added code for submission of comments

File: packages/web/components/CommentBoxes.tsx
Comment (outlawhayden) [inline packages/web/components/CommentBoxes.tsx:1]: more comment box code

File: packages/web/components/Rubric.tsx
Comment (outlawhayden) [inline packages/web/components/Rubric.tsx:1]: rubric code for submission rating on each card

File: packages/web/components/ThreeCardLayout.tsx
Comment (outlawhayden) [inline packages/web/components/ThreeCardLayout.tsx:1]: three card response layout for rating page

File: packages/web/lib/api.ts
Comment (outlawhayden) [inline packages/web/lib/api.ts:1]: adding 'varied' response type to api

File: packages/googlecloud/functions/getanswer/inquirer.py [hunk 1]
```diff
@@ -192,18 +159,65 @@ def get_indepth_response_from_query(df, db, query, k):
...
+    responses_llm = chain_llm.run(
+            question=query, docs=docs_page_content, temperature=0
+        )
+
+    return process_responses_llm(responses_llm, docs, query_type)
+
+## varied responses for user annotation
+def get_varied_response_from_query(df, db, query, k, n = 1, card_type = "varied"):
+    logger.info("Performing varied summary query...")
+    master_response = {}
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/inquirer.py:176]: suggestion: replace with `main_response`

[Reasoning](https://github.com/github/renaming)

File: packages/googlecloud/functions/getanswer/inquirer.py [hunk 2]
```diff
@@ -192,18 +159,65 @@ def get_indepth_response_from_query(df, db, query, k):
...
+    master_response["card_type"] = "varied"
+    response_list = {}
+    k_list = [3,5,7]
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/inquirer.py:179]: Why suck small k values? Also not a big difference between them? My gut would be 6, 12, 25.

File: packages/googlecloud/functions/getanswer/main.py [hunk 3]
```diff
@@ -79,44 +46,28 @@ def getanswer(request):
 
     # Set CORS headers for the main request
     headers = {"Access-Control-Allow-Origin": "*"}
-
     logging.info(f"getanswer {API_VERSION}")
     start = time.time()
 
     # Parse args
+    print(request.headers)
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/main.py:53]: Remove print statements

File: packages/googlecloud/functions/getanswer/requirements.txt [hunk 4]
```diff
@@ -3,8 +3,8 @@ functions-framework
 flask
 google-cloud-error-reporting
 python-dotenv
-langchain==0.0.330
-openai==0.28.1
+langchain
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/requirements.txt:6]: Would also keep thi file as-is

File: packages/transcription/transcribe/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+## TU Capstone- Transcription
```
Comment (marvinmarnold) [inline packages/transcription/transcribe/README.md:1]: Very cool!

File: packages/transcription/transcribe/oauth.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import os
+from dotenv import load_dotenv
+
+def import_env_vars():
+    os.chdir(r"packages\transcription")
+    load_dotenv(r"cred\cred.env")
+
+    # Get credentials from environment variables
+    YOUTUBE_API_KEY = os.getenv("YOUTUBE_API_KEY")
+    CLIENT_ID = os.getenv("CLIENT_ID")
```
Comment (marvinmarnold) [inline packages/transcription/transcribe/oauth.py:11]: Where do CLIENT_ID and CLIENT_SECRET come from?

File: packages/web/app/api/v1/cards/like.ts [hunk 7]
```diff
@@ -1,60 +1,70 @@
...
 "use client";
 
-import { TABLES } from "@/lib/supabase/db";
-import { createClient } from "@supabase/supabase-js";
-import { NextResponse } from "next/server";
+import { TABLES } from "@/lib/supabase/db"; // Importing table definitions
+import { createClient } from "@supabase/supabase-js"; // Supabase client for interacting with the database
+import { NextResponse } from "next/server"; // Next.js server-side response object
 
+// Environment variables for Supabase URL and secret service key
```
Comment (marvinmarnold) [inline packages/web/app/api/v1/cards/like.ts:8]: Appreciate the comments!

File: packages/web/app/error.tsx [hunk 8]
```diff
@@ -1,10 +0,0 @@
-"use client";
```
Comment (marvinmarnold) [inline packages/web/app/error.tsx:1]: This file should be kept

File: packages/web/app/not-found.tsx [hunk 9]
```diff
@@ -1,7 +0,0 @@
-"use client";
```
Comment (marvinmarnold) [inline packages/web/app/not-found.tsx:1]: This file should be kept

File: packages/web/app/page.tsx [hunk 10]
```diff
@@ -1,12 +1,7 @@
-import { getPageMetadata } from "@/lib/paths";
 import HomeBanner from "../components/HomeBanner";
 import HomeLearnMore from "../components/HomeLearnMore";
 import HomeResults from "../components/HomeResults";
-
 export const dynamic = "force-dynamic";
-
-export const metadata = getPageMetadata("Eye on Surveillance");
```
Comment (marvinmarnold) [inline packages/web/app/page.tsx:8]: Keep this logic

File: packages/web/app/s/[cardId]/page.tsx [hunk 11]
```diff
@@ -1,24 +1,7 @@
 import BetaCard from "@/components/BetaCard";
-import { getPageMetadata } from "@/lib/paths";
 import { TABLES } from "@/lib/supabase/db";
 import { supabase } from "@/lib/supabase/supabaseClient";
 
-export async function generateMetadata({
```
Comment (marvinmarnold) [inline packages/web/app/s/[cardId]/page.tsx:6]: Keep this

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Awesome to see all this. The biggest issue I see is that it overwrites several of the recent changes Ayyub/I made. I'd probably recommend that you start a new branch off of the current version of main. Then manually copy over your relevant changes.

Could you also add `.DS_Store` to gitignore and make sure it is not added to git?

**Comment by outlawhayden** [comment]
just dumped the ds_store - our bad. will start pulling everything apart and back together this afternoon and tomorrow

File: packages/transcription/transcribe/README.md (hunk 5 above)
Comment (outlawhayden) [reply packages/transcription/transcribe/README.md:1]: thanks! it's not 100% working, but as we proceed with building other little features we're going to open this back up and start using it more

File: packages/transcription/transcribe/oauth.py (hunk 6 above)
Comment (outlawhayden) [reply packages/transcription/transcribe/oauth.py:11]: @mikafur32 is the one who built this - i believe a config file but i could be wrong

File: packages/web/app/api/v1/cards/like.ts (hunk 7 above)
Comment (outlawhayden) [reply packages/web/app/api/v1/cards/like.ts:8]: thanks!

File: packages/web/components/Citation.tsx [hunk 12]
```diff
@@ -1,9 +1,3 @@
-import {
```
[the remark this answers is not in the document]
Comment (outlawhayden) [reply packages/web/components/Citation.tsx:1]: our side or your side?

File: packages/web/app/error.tsx [hunk 13]
```diff
@@ -7,4 +7,4 @@ export default function Error({}: {
   reset: () => void;
 }) {
   return <GenericError />;
-}
+}
```
Comment (outlawhayden) [inline packages/web/app/error.tsx:10]: file restored

File: packages/web/app/not-found.tsx [hunk 14]
```diff
@@ -4,4 +4,4 @@ import GenericError from "@/components/GenericError";
 
 export default function NotFound() {
   return <GenericError />;
-}
+}
```
Comment (outlawhayden) [inline packages/web/app/not-found.tsx:7]: file restored

File: packages/googlecloud/functions/getanswer/main.py
Comment (outlawhayden) [reply packages/googlecloud/functions/getanswer/main.py:1]: i agree - we don't do any actual getanswer stuff live, this is just to run locally so we can create the data for the tool. might just need to rebuild our bit here 100%

File: packages/backend/src/preprocessor.py [hunk 15]
```diff
@@ -102,23 +113,14 @@ def create_db_from_news_transcripts(news_json_directory):
 
         data = loader.load()
         text_splitter = RecursiveCharacterTextSplitter(
-            chunk_size=10000, chunk_overlap=5000
+            chunk_size=500, chunk_overlap=250
```
Comment (marvinmarnold) [inline packages/backend/src/preprocessor.py:116]: @ayyubibrahimi does this look like a regression to you? @outlawhayden any reason for the change?

File: packages/backend/src/preprocessor.py [hunk 16]
```diff
@@ -135,19 +137,10 @@ def create_db_from_cj_transcripts(cj_json_directory):
 
         data = loader.load()
         text_splitter = RecursiveCharacterTextSplitter(
-            chunk_size=15000, chunk_overlap=7500
+            chunk_size=20000, chunk_overlap=10000
```
Comment (marvinmarnold) [inline packages/backend/src/preprocessor.py:140]: Here too @ayyubibrahimi  @outlawhayden

File: packages/backend/src/preprocessor.py [hunk 17]
```diff
@@ -198,7 +185,7 @@ def create_db_from_public_comments(pc_json_directory):
 
         data = loader.load()
         text_splitter = RecursiveCharacterTextSplitter(
-            chunk_size=10000, chunk_overlap=5000
+            chunk_size=1000, chunk_overlap=500
```
Comment (marvinmarnold) [inline packages/backend/src/preprocessor.py:188]: @ayyubibrahimi @outlawhayden

File: packages/googlecloud/functions/getanswer/helper.py [hunk 18]
```diff
@@ -38,31 +36,30 @@ def get_dbs():
...
-    llm = ChatOpenAI(model="gpt-4")
-
-    base_embeddings = OpenAIEmbeddings()
+    llm = OpenAI()
 
     general_prompt_template = """
-    As an AI assistant, your role is to provide concise, balanced summaries from the transcripts of New Orleans City Council meetings in response to the user's query "{user_query}". Your response should not exceed one paragraph in length. If the available information from the transcripts is insufficient to accurately summarize the issue, respond with 'Insufficient information available.' If the user's query extends beyond the scope of information contained in the transcripts, state 'I don't know.'
+    As an AI assistant tasked with generating brief general summaries, your role is to provide succinct, balanced information from the transcripts of New Orleans City Council meetings in response to the question "{question}". The response should not exceed one paragraph in length. If the available information from the transcripts is insufficient to accurately summarize the issue, please respond with 'Insufficient information available.' If the question extends beyond the scope of information contained in the transcripts, state 'I don't know.'
```
Comment (marvinmarnold) [inline packages/googlecloud/functions/getanswer/helper.py:42]: @ayyubibrahimi which version is preferred?

File: packages/web/.env [hunk 19]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (marvinmarnold) [inline packages/web/.env:1]: This shouldn't be in source control

File: packages/web/app/tips/page.tsx [hunk 20]
```diff
@@ -10,8 +9,6 @@ const Step = ({ title, text }: { title: string; text: string }) => {
   );
 };
 
-export const metadata = getPageMetadata("Tips");
```
Comment (marvinmarnold) [inline packages/web/app/tips/page.tsx:13]: Should be kept

File: packages/web/components/Cards.tsx [hunk 21]
```diff
@@ -1,16 +0,0 @@
-import { ICardOld } from "@/lib/api";
```
Comment (marvinmarnold) [inline packages/web/components/Cards.tsx:1]: I think this should be kept, but I'm not sure. Generally there's probably not usually a reason for this PR to delete things.

File: packages/web/components/Citation.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (marvinmarnold) [reply packages/web/components/Citation.tsx:1]: EOS side (left side). The iframe is what makes youtube videos embedded.

File: packages/web/components/GenericError.tsx [hunk 22]
```diff
@@ -1,22 +0,0 @@
-"use client";
```
Comment (marvinmarnold) [inline packages/web/components/GenericError.tsx:1]: Keep

File: packages/web/lib/paths.ts [hunk 23]
```diff
@@ -2,31 +2,10 @@ export const ABOUT_BETA_PATH = "/about";
 export const TIPS_PATH = "/tips";
 export const CARD_SHOW_PATH = "/s";
 export const HOME_PATH = "/";
+export const USER_FEEDBACK = "/feedback";
 
-export const API_NEW_CARD_PATH = "/api/v1/cards";
```
Comment (marvinmarnold) [inline packages/web/lib/paths.ts:6]: Keep

**Pull request closed without merging**